forum/profiles/views.py

```diff
@@ -1,3 +1,17 @@
-from django.shortcuts import render
+from rest_framework.viewsets import ModelViewSet
+from rest_framework.permissions import IsAuthenticated
+from .models import InvestorProfile
+from .serializers import InvestorProfileSerializer
+from .permissions import IsOwnerOrReadOnly
 
-# Create your views here.
+class InvestorViewSet(ModelViewSet):
+
+    queryset = InvestorProfile.objects.all()
+    serializer_class = InvestorProfileSerializer
+    permission_classes = [IsAuthenticated, IsOwnerOrReadOnly]
+
+    def get_queryset(self):
+        return InvestorProfile.objects.filter(user=self.request.user)
```

Comment:
Use `select_related` to prefetch related user objects for efficiency when accessing the user attributes.
```
def get_queryset(self):
    return InvestorProfile.objects.filter(user=self.request.user).select_related('user')
```

Reply:
Done